kernel/src/process/posix_thread/futex.rs

```diff
@@ -90,13 +112,15 @@ pub fn futex_requeue(
     max_nwakes: usize,
     max_nrequeues: usize,
     futex_new_addr: Vaddr,
+    pid: Pid,
+    flags: FutexFlags,
```

Comment:
Can we use only one parameter `pid: Option<Pid>` here instead of having both `pid` and `flags`? It seems the flags is redundant.